File: chat_gpt_aws_lambda/bedrock_cdk/src/lambda/bedrock/handler.py
import json

def bedrockHandler(event, context=None):
  print('in bedrockHandler')
  print(event)
  http_method = event['httpMethod']
  query_parameters = event['queryStringParameters']
  print(http_method, query_parameters)
  response = {
    "statusCode":200,
    "body":json.dumps({
      "message":"hello python lambda",
      "httpMethod":http_method,
      "query_parameters":query_parameters
      })
  }

File: chat_gpt_aws_lambda/chatgpt_cdk/src/lambdas/handler.ts
import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
import { HTTP_CODE, HTTP_METHOD, jsonApiProxyResultResponse } from '../util';
import { ChatGptHandler } from './chatgpt/ChatGptHandler';
import { BedrockHandler } from './bedrock/BedrockHandler';

export const handler = async (
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> => {
  const api = new ChatGptHandler(event);
  switch (event.httpMethod) {
    case HTTP_METHOD.GET:
      return await api.get();
    case HTTP_METHOD.POST:
      return await post(event);
    default:
      return await api.noService();
  }
};

const post = async (
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> => {
  if (!event.body)
    return jsonApiProxyResultResponse(HTTP_CODE.NOT_FOUND, {
      message: false,
      body: 'missing body',
    });

  const { engine } = JSON.parse(event.body);
  if (engine.startsWith('chat')) {
    const api = new ChatGptHandler(event);
    return await api.post();
  } else {
    const api = new BedrockHandler(event);
    return await api.post();
  }
};
